XSIAM CyberArkEPV ModelingRules (#27932)
demisto/content: Packs/CyberArkEPV/README.md

# CyberArk Enterprise Password Vault
This pack includes Cortex XSIAM content.

## Configuration on Server Side
Syslog messages can be sent to multiple syslog servers in two different ways.
- One message can be sent to multiple servers by configuring an XSLT file.

> - Multiple messages can be sent to different servers and formatted differently for each server by configuring multiple XSLT files, formats, and code-message lists. The code-message lists must be matched. They must contain the same number of items in the same order.

Reviewer: ```suggestion
- Multiple messages can be sent to different servers and formatted differently for each server by configuring multiple XSLT files, formats, and code-message lists. The code-message lists must contain the same number of items in the same order.
```

```http

>   Note: The .ini file contains these configuration values.

Reviewer: Not sure what configuration values you are referring to here. Do you mean the values in Step 1?

Author: Hi Shirley,
Yes

```
1. In \PrivateArk\Server\DBParm.sample.ini, copy the SYSLOG section.
- **SyslogServerIP** — The IP addresses of the Syslog servers where messages are sent. Specify multiple values with commas.
- **SyslogServerProtocol** — Specifies the Syslog protocol that is used to send audit logs. Specify TCP or UDP. The default value is UDP.
- **SyslogServerPort** — The port used to connect to the Syslog server. The default value is 514.
- **SyslogMessageCodeFilter** — Defines which message codes are sent from the Vault to the BrokerVM through the Syslog protocol. You can specify message numbers or ranges of numbers, separated by commas. Specify multiple values with pipelines. By default, all message codes are sent for user and safe activities.
- **SyslogTranslatorFile** — Specifies the XSL file used to parse CyberArk audit records data into Syslog protocol. Specify multiple values with commas.

> - **DebugLevel** — Determines the level of debug messages. Specify SYSLOG(2) to include Syslog xml messages in the trace file.

Reviewer: ```suggestion
- **DebugLevel** — Determines the level of debug messages. Specify SYSLOG(2) to include Syslog XML messages in the trace file.
```

- **UseLegacySyslogFormat** — Controls the format of the syslog message, and defines whether it is sent in a newer syslog format (RFC 5424) or in a legacy format. The default value is No, which enables working with the newer syslog format. Specify multiple values with commas.

> 2. In DBParm.ini, paste the SYSLOG section at the bottom of the file, then rename the file to XSIAM.xsl.

Reviewer: Is this the same file as in step 1?

3. Copy the relevant XSL translator file from the syslog subfolder of the server installation folder to the location specified
in the SyslogTranslatorFile parameter in DBParm.ini.
```http
Note: During vault installation or upgrade, sample XSL files are copied to the PrivateArk\Server\syslog folder.
```

4. Make any needed changes to the XSL translator file relevant to BrokerVM implementation.
5. Stop and Start the vault for the changes to take effect.

## Collect Events from Vendor
In order to use the collector, use the [Broker VM](#broker-vm) option.

### Broker VM
To create or configure the Broker VM, use the information described [here](https://docs-cortex.paloaltonetworks.com/r/Cortex-XDR/Cortex-XDR-Pro-Administrator-Guide/Configure-the-Broker-VM).

You can configure the specific vendor and product for this instance.

1. Navigate to **Settings** > **Configuration** > **Data Broker** > **Broker VMs**. 
2. Go to the apps tab and add the **Syslog** app. If it already exists, click the **Syslog** app and then click **Configure**.

> 3. Click **Add New**.

Reviewer: After this step, don't you need to enter the vendor and product name?  
Are there any additional steps after that?

Author: No, the vendor and product will be auto-filled.

#### Set up the Syslog Configuration
1. Navigate to the *Conf* subfolder under the CyberArk Vault server installation folder (*PrivateArk\Server\Conf*).
2. Copy the *\[SYSLOG\]* section from the *DBParm.sample.ini* sample file, and paste it at the bottom of the *DBParm.ini* file. 
3. Set the following parameters under the copied *\[SYSLOG\]* section in the *DBParm.ini* file
   | Parameter                       | Description    
   | :---                            | :---                    
   | `SyslogServerIP`                | IP address of the Cortex XSIAM Broker VM Syslog Server.  
   | `SyslogServerPort`              | Target port that the Cortex XSIAM Broker VM Syslog Server is listening on for receiving Syslog messages from Cyber-Ark.  
   | `SyslogServerProtocol`          | The protocol that will be used to forward the Syslog messages to Cortex XSIAM: *UDP* (the default setting), *TCP* or *TLS* (Note: for *TLS*, additional settings are required for configuring certificates, see [*Configure encrypted and non-encrypted protocols*](https://docs.cyberark.com/PAS/Latest/en/Content/PASIMP/Integrating-with-SIEM-Applications.htm#Configureencryptedandnonencryptedprotocols)).

>    | `SyslogMessageCodeFilter`       | Range or list of requested message codes that should be sent to  Cortex XSIAM through the syslog protocol. See [*Vault Audit Action Codes*](https://docs.cyberark.com/PAS/Latest/en/Content/PASREF/Vault%20Audit%20Action%20Codes.htm) for the complete list of vault events message codes. By default, all message codes are sent for user and safe activities. For including all Vault events, define the following range: *0-999*. 

Reviewer: ```suggestion
   | `SyslogMessageCodeFilter`       | Range or list of requested message codes that should be sent to Cortex XSIAM through the syslog protocol. See [*Vault Audit Action Codes*](https://docs.cyberark.com/PAS/Latest/en/Content/PASREF/Vault%20Audit%20Action%20Codes.htm) for the complete list of vault event message codes. By default, all message codes are sent for user and safe activities. For including all Vault events, define the following range: *0-999*. 
```

>    | `SyslogTranslatorFile`   | Specify the relative path in the Cyber-Ark Vault server installation folder (*PrivateArk\Server*) to the relevant XLS translator file  (see [*Set up the XSL Translator*](#Set-up-the-XSL-Translator) section above). For example: *Syslog\XSIAM.xsl*.

Reviewer: ```suggestion
   | `SyslogTranslatorFile`   | Specify the relative path in the CyberArk Vault server installation folder (*PrivateArk\Server*) to the relevant XLS translator file  (see [*Set up the XSL Translator*](#Set-up-the-XSL-Translator) section above). For example: *Syslog\XSIAM.xsl*.
```

>    | `UseLegacySyslogFormat`   | Controls whether the syslog messages should be sent in the old legacy syslog format (*Yes*), or in the newer modern [RFC 5424](https://datatracker.ietf.org/doc/html/rfc5424) format (*No*). For Cortex XSIAM set this parameter with the default value of *No*. 

Reviewer: ```suggestion
   | `UseLegacySyslogFormat`   | Controls whether the syslog messages should be sent in the old legacy syslog format (*Yes*), or in the newer modern [RFC 5424](https://datatracker.ietf.org/doc/html/rfc5424) format (*No*). For Cortex XSIAM, set this parameter with the default value of *No*. 
```

>    |`SendMonitoringMessage`| Controls whether the Syslog messages that are sent to Cortex XSIAM should include periodic server* system monitoring* events as well (in addition to *audit events*). For Cortex XSIAM set this parameter with the default value of *no*.

Reviewer: ```suggestion
   |`SendMonitoringMessage`| Controls whether the syslog messages that are sent to Cortex XSIAM should include periodic server *system monitoring* events as well (in addition to *audit events*). For Cortex XSIAM, set this parameter with the default value of *No*.
```

 See [*DBPARM.ini file parameters*](https://docs.cyberark.com/PAS/Latest/en/Content/PASIMP/Integrating-with-SIEM-Applications.htm#DBPARMinifileparameters) for a complete list of the possible *DBPARM.ini* file syslog parameters.

>  Bellow is a sample *\[SYSLOG\]* configuration section for the *DBParm.ini* file: 

Reviewer: ```suggestion
 Below is a sample *\[SYSLOG\]* configuration section for the *DBParm.ini* file: 
```

 
  ```BASH        
            [SYSLOG]
            SyslogServerIP=192.168.1.123
            SyslogServerPort=514
            SyslogServerProtocol=UDP
            SyslogMessageCodeFilter=0-999
            SyslogTranslatorFile=Syslog\XSIAM.xsl
            UseLegacySyslogFormat=No
            SendMonitoringMessage=no
``` 
4. Restart the Vault server to apply the configuration changes. 

### Remarks

> Cyber-Ark Vault supports additional Syslog configuration settings such as forwarding audit events to *multiple* Syslog servers, each server with it's own unique set of syslog parameters. For additional details refer to the [Cyber-Ark Vault documentation](https://docs.cyberark.com/PAS/Latest/en/Content/PASIMP/Integrating-with-SIEM-Applications.htm?tocpath=End%20user%7CReports%20and%20Audits%7C_____6).

Reviewer: ```suggestion
CyberArk Vault supports additional syslog configuration settings such as forwarding audit events to *multiple* syslog servers, each server with it's own unique set of syslog parameters. For additional details, refer to the [Cyber-Ark Vault documentation](https://docs.cyberark.com/PAS/Latest/en/Content/PASIMP/Integrating-with-SIEM-Applications.htm?tocpath=End%20user%7CReports%20and%20Audits%7C_____6).
```

## Collect Events from Vendor
In order to use the collector, use the [Broker VM](#broker-vm) option.

### Broker VM
To create or configure the Broker VM, use the information described [here](https://docs-cortex.paloaltonetworks.com/r/Cortex-XDR/Cortex-XDR-Pro-Administrator-Guide/Configure-the-Broker-VM).

1. Navigate to **Settings** > **Configuration** > **Data Broker** > **Broker VMs**. 
2. Go to the apps tab and add the **Syslog** app. If it already exists, click the **Syslog** app and then click **Configure**.
3. Click **Add New**.
4. Set the following parameters for the Syslog configuration:
   | Parameter     | Value    
   | :---          | :---                    

>    | `Protocol`    | Should be aligned with the protocol defined in the *SyslogServerProtocol* parameter in the `[SYSLOG]` section of the *DBParm.ini* configuration file on the Cyber-Ark Vault server (see [Set up the Syslog Configuration](#Set-up-the-Syslog-Configuration)).   

Reviewer: ```suggestion
   | `Protocol`    | Should be aligned with the protocol defined in the *SyslogServerProtocol* parameter in the `[SYSLOG]` section of the *DBParm.ini* configuration file on the CyberArk Vault server (see [Set up the Syslog Configuration](#Set-up-the-Syslog-Configuration)).   
```

>    | `Port`        | Should be aligned with the protocol defined in the *SyslogServerPort* parameter in the `[SYSLOG]` section of the *DBParm.ini* configuration file on the Cyber-Ark Vault server (see [Set up the Syslog Configuration](#Set-up-the-Syslog-Configuration)).   

Reviewer: ```suggestion
   | `Port`        | Should be aligned with the protocol defined in the *SyslogServerPort* parameter in the `[SYSLOG]` section of the *DBParm.ini* configuration file on the CyberArk Vault server (see [Set up the Syslog Configuration](#Set-up-the-Syslog-Configuration)).   
```

>    | `Format`      | Select **_CEF_**. 

Reviewer: ```suggestion
   | `Format`      | Select **CEF**. 
```
